Power Up Proto/src/org/usfirst/frc/team5493/robot/subsystems/OneClimbyBoi.java

```diff
@@ -5,66 +5,106 @@
 
 import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
 
-import edu.wpi.first.wpilibj.CounterBase.EncodingType;
-import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.Counter;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.Preferences;
 import edu.wpi.first.wpilibj.command.Subsystem;
 
 /**
  *
  */
 public class OneClimbyBoi extends Subsystem {
 
-	public WPI_TalonSRX climbMotor;
-	public Object climbDown;
-	
-	public static final double kDistancePerRevolution = 18.84;
-	public static final double kPulsesPerRevolution = 1024;
-	public static final double kDistancePerPulse = kDistancePerRevolution / kPulsesPerRevolution;
+	//TODO: Hold can be removed when PID is integrated.
 	
 	//Encoder numbers to be added
 	//Encoder A Channel -> DIO 1, Module 1;
 	
 	//private Encoder encoder = new Encoder(1, 2, false, EncodingType.k2X);
 	
 	
+	private WPI_TalonSRX climbMotor;
+
+	private double downSpeed;
+	private double upSpeed;
+	private double holdSpeed;
+
+	private DigitalInput climbuplimitSwitch = new DigitalInput(RobotMap.LIMIT_SWITH_UP_CLIMBER);
+	private Counter climbupCounter = new Counter(climbuplimitSwitch);
+	private DigitalInput climbdownlimitSwitch = new DigitalInput(RobotMap.LIMIT_SWITH_DOWN_CLIMBER);
+	private Counter climbdownCounter = new Counter(climbdownlimitSwitch);
+	
+	public static final double kDistancePerRevolution = 18.84;
+	public static final double kPulsesPerRevolution = 1024;
+	public static final double kDistancePerPulse = kDistancePerRevolution / kPulsesPerRevolution;
+
+
+	public OneClimbyBoi() {
+		super();
+
+		climbMotor = new WPI_TalonSRX(RobotMap.CLIMBER);
+		// Put methods for controlling this subsystem
+		// here. Call these from Commands.
+		double secondsFromNeutral = 0;
+		int timeoutMs = 0;
+
+		Preferences prefs = Preferences.getInstance();
+
+		secondsFromNeutral = prefs.getDouble("RampRateClimber", 0.25);
+		upSpeed = prefs.getDouble("ClimberUpSpeed", 0.8);
+		downSpeed = prefs.getDouble("ClimberDownSpeed", -0.4);
+		holdSpeed = prefs.getDouble("ClimberHoldSpeed", 0.1);
+		timeoutMs = prefs.getInt("RampRateClimberTimeout", 1);
+
+		climbMotor.configOpenloopRamp(secondsFromNeutral, timeoutMs);
+	}
+
+	public void initDefaultCommand() {
+		// Set the default command for a subsystem here.
+		// setDefaultCommand(new MySpecialCommand());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)